### DISCUSSION TITLE
No way to disable prefetching completely in next/link

### DISCUSSION THREAD

**Discussion by mmmulani** [vercel/next.js#39577] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Binaries:
      Node: 18.5.0
      npm: 8.12.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 12.2.3
      eslint-config-next: 11.0.1
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

While you can disable prefetching when a `<Link>` comes in the viewport, there is no way to disable prefetching on hover with `<Link>`.

This is a bit frustrating for us as we show many links on the page, and it is reasonable that a user could hover over 10-50 before clicking on one. Also, the prefetching use of `<Link>` is irrelevant to us as we prefer to do client-side fetching.

On the flip side though, `<Link>` is great at doing client-side navigation for us, and I prefer using that over manually messing with `router.push/replace` myself.

This behaviour was added in #18859 but sadly there was no discussion about this case.

### Expected Behavior

Would be great if `prefetch={false}` disabled prefetching on hover.

### Link to reproduction

https://github.com/vercel/next.js/blob/canary/packages/next/client/link.tsx#L477-L496

### To Reproduce

Create a `<Link prefetch={false} />` and see that this code is evaluated when the element is hovered on: https://github.com/vercel/next.js/blob/canary/packages/next/client/link.tsx#L477-L496

**Comment by mmmulani** [comment]
cc @Timer original creator of that PR

**Comment by joaoGabriel55** [comment]
Hey @mmmulani I can take a look on it!

**Comment by DaleWebb** [comment]
@mmmulani - Have you considered making your own link component by using the `useRouter` hook?

**Comment by mmmulani** [comment]
I looked into it but `<Link>` actually has tons of useful logic that I don't really want to replicate/mantain, see here: https://github.com/vercel/next.js/blob/canary/packages/next/client/link.tsx#L109-L162

on the flip side, it seems like adding this functionality to the component would be pretty easy, it already has a `shouldPrefetch` value. Though, I'm not sure it's what the API designers would want, so I'm moreso waiting for their input before creating a PR
